### ISSUE TITLE
Forward sync works solely when no pdf pane was open before

### ISSUE THREAD

**Issue by karyon** [thomasjo/atom-latex#209]
when building the project when no pdf pane is open, it starts compiling and then after a while, opens the pdf pane and jumps to the location that corresponds to my cursor in the .tex file, so that works as expected. backwards sync (clicking into the pdf) works as well.

however, when changing the cursor's location and then building again, the pdf view stays at the same location as it was before. as i understand #170, this should work.

also, i'd expect something like (right-click in tex file -> jump to location in pdf), or a hotkey like ctrl-something that jumps to the current cursor's location, but that's definitely a feature request (albeit a massively useful one, at least for me).

basically, right now there is no reasonable way to get from a tex file to the corresponding location in the pdf, i always need to manually scroll and search through the pdf.

**Comment by thomasjo** [comment]
In order to help figure this out, we're going to need more information; operating system, LaTeX distribution and version (and if you're using MiKTeX, we'll need to know if you're using the latexmk or texify builder), as well as which PDF viewer you're using.

This functionality works as expected when using TeX Live 2016 on macOS 10.11.6 together with Skim, Preview and atom-pdf, as well as other configurations.

As for the context menu enhancement, that's likely something we can implement if we can limit it using grammar scopes. Ctrl-clicking is not going to happen anytime soon as there's no supported way to use mouse clicks in Atom keymaps, and any other technique to support this functionality would simply be too invasive (and guaranteed to collide with other packages using such invasive techniques).

**Comment by karyon** [comment]
Windows 10 64bit, atom 1.11.0-beta3, miktex 2.9.6050, texify, xelatex, atom pdf-view. i copied synctex.exe from a texlive installation since miktex does not ship this to make backwards sync work.

it doesn't work for me with any combination of [texify|latexmk][pdflatex|xetex].

however it does work when using sumatraPDF as viewer.

i'd actually prefer a hotkey (that's what i meant with ctrl-something :) ) instead of ctrl-click, maybe that's also easier to implement... as i see it it's just ctrl-alt-b without building, but of course i don't know what else is happening behind the scenes.

**Comment by yitzchak** [comment]
I have reproduced this. Basically sync (<kbd>ctrl</kbd>+<kbd>alt</kbd>+<kbd>s</kbd>) does not work on Windows with pdf-view if the pdf is already open. It *does* work on Linux.

**Comment by karyon** [comment]
i wasn't aware of ctrl-alt-s, thanks! seems like it's mentioned nowhere except in the keybindings in the settings...

**Comment by yitzchak** [comment]
pdf-view does not display any error information if it fails to run `synctex`. I believe that this what is happening. `synctex` from TeX Live depends on the kpathsea library through `kpathsea622.dll`. If this dll is not in the system path then execution will fail. Trying to run `synctex` from the command line will show this. I was able to get the SyncTeX functionality to work with TeX Live, but it breaks when I remove TeX Live from the path and just copy `synctex` without `kpathsea622.dll`. When I copy `kpathsea622.dll` it starts working with MiKTeX.

The real issue appears to me that MiKTeX is assuming that SyncTeX parsing code will be located inside the editor (TeXworks does it this way) whereas pdf-view assumes that the `synctex` executable is available to take of the parsing. The source code for `synctex` is in the MiKTeX repo, but they are not compiling it in their build.

@thomasjo Do you think we should have a warning message that SyncTeX forward does not work in the case MiKTeX and pdf-view are in use?

**Comment by karyon** [comment]
it seems i've been confusing things. i have a system where i installed the whole texlive and am using the synctex from there, and in fact ctrl-alt-s is always working there (following your explanation it's because the dll is there besides the .exe)

on the system where i copied synctex, it is not working at all. calling the exe directly tells me the dll is missing.

however, what still seems broken on the first system is the automatic sync after building when the pdf is open.

to clarify: if pdf-view is not open, ctrl-alt-b builds the files, opens pdf-view, and syncs. changing the cursors location and pressing ctrl-alt-s syncs. changing the cursors location again and pressing ctrl-alt-b rebuilds and reloads the pdf, but does not sync, it stays at the same position.

also, i'd suggest to find a more prominent place for explaining ctrl-alt-s (such as the README usage section), because if i had known about it, this bug report would've been a lot more straightforward :)

**Comment by yitzchak** [comment]
Okay. I understand. I have not been able to replicate that behavior yet. Did you makes changes in the file before you did the rebuild?

**Comment by karyon** [comment]
alright, some more testing. when using latexmk and making no modifications, ctrl-alt-b takes about half a second and then syncs. when doing any changes, it takes longer and does not sync. with texify, it takes quite a bit of time even with no changes, and never syncs afterwards.

i debugged this a bit and it seems like the forward-sync method of the pdf viewer is always called.

the only thing i found out that when i place a breakpoint into the `console.log('Warning: ' + msg);` line in pdf.worker.js (I get a bunch of `Warning: Badly formated number`, and press continue a few times until it's through all the warnings, then it works. some race condition between the syncing and reloading of the file? unfortunately i have no idea what's going on and how i would further debug this...

**Comment by thomasjo** [comment]
Hmm, seems like a bug in pdf-view's sync implementation. I'll see if I can reproduce in a VM while throttling its compute resources, etc.

**Comment by karyon** [comment]
i'm on a desktop ivy bridge i5, if that helps.

maybe the loading times of the document also play a role? my document is about 35 pages, almost no graphics. it's available here: https://github.com/karyon/masterthesis

**Comment by thomasjo** [comment]
Thanks for the extra info 🙇

**Comment by karyon** [comment]
one more observation, i just saw that after rebuilding, the correct location flashes up shortly before the old (wrong location) is set and stays there. now that i noticed it, this is reliably reproducible for me.

**Comment by alexander-myltsev** [comment]
Hi,

System info: osx 10.11.6, atom 1.11.1, latex 0.37.1, TeXLive-2015, pdf-view 0.50.0

Sync from PDF to tex source works just fine. Vice versa sync, when I press Ctrl-Alt+S, for some reason (despite `latex atom` settings) gets me to DVI. What should I switch to make it work properly?

**Comment by yitzchak** [comment]
@alexander-myltsev What are the contents of the `foo.fdb_latexmk` file? Where, `foo` is the name of your main LaTeX file?

**Comment by alexander-myltsev** [comment]
```
# Fdb version 3
["pdflatex"] 1476427431 "jfp2egui.tex" "jfp2egui.pdf" "jfp2egui" 1476427432
  "/usr/local/texlive/2015/texmf-dist/fonts/enc/dvips/base/8r.enc" 1165713224 4850 80dc9bab7f31fb78a000ccfed0e27cab ""
  "/usr/local/texlive/2015/texmf-dist/fonts/map/fontname/texfonts.map" 1272929888 3287 e6b82fe08f5336d4d5ebc73fb1152e87 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/symbol/psyr.tfm" 1136768653 1408 5937f58aa508ea2cea4901c07d10f5fe ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/psyro.tfm" 1136768653 1544 23a042a74981a3e4b6ce2e350e390409 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmb7t.tfm" 1136768653 2172 fd0c924230362ff848a33632ed45dc23 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmb8r.tfm" 1136768653 4524 6bce29db5bc272ba5f332261583fee9c ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmbi7t.tfm" 1136768653 2228 e564491c42a4540b5ebb710a75ff306c ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmbi8r.tfm" 1136768653 4480 10409ed8bab5aea9ec9a78028b763919 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmr7t.tfm" 1136768653 2124 2601a75482e9426d33db523edf23570a ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmr8r.tfm" 1136768653 4408 25b74d011a4c66b7f212c0cc3c90061b ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmrc7t.tfm" 1136768653 2680 312a2d12b1f1df8ee0212e7ba1962402 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmri7t.tfm" 1136768653 2288 f478fc8fed18759effb59f3dad7f3084 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/ptmri8r.tfm" 1136768653 4640 532ca3305aad10cc01d769f3f91f1029 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/zptmcm7m.tfm" 1136768653 2232 db256afffc8202da192b4641df14d602 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/zptmcm7t.tfm" 1136768653 2172 1d00c2a0d10f23031be62329457a870c ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/zptmcm7v.tfm" 1136768653 1032 20febbd0f0c9a48eb78616f897008286 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/adobe/times/zptmcm7y.tfm" 1136768653 1520 ad7b3c1a480a03b3e41b5fbb13d938f2 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/amsfonts/symbols/msam10.tfm" 1246382020 916 f87d7c45f9c908e672703b83b72241a3 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/amsfonts/symbols/msam5.tfm" 1246382020 924 9904cf1d39e9767e7a3622f2a125a565 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/amsfonts/symbols/msam7.tfm" 1246382020 928 2dc8d444221b7a635bb58038579b861a ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/amsfonts/symbols/msbm10.tfm" 1246382020 908 2921f8a10601f252058503cc6570e581 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/amsfonts/symbols/msbm5.tfm" 1246382020 940 75ac932a52f80982a9f8ea75d03a34cf ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/amsfonts/symbols/msbm7.tfm" 1246382020 940 228d6584342e91276bf566bcf9716b83 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmmi10.tfm" 1136768653 1528 abec98dbc43e172678c11b3b9031252a ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmmi6.tfm" 1136768653 1512 f21f83efb36853c0b70002322c1ab3ad ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmmi9.tfm" 1136768653 1524 d89e2d087a9828407a196f428428ef4a ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmr10.tfm" 1136768653 1296 45809c5a464d5f32c8f98ba97c1bb47f ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmr6.tfm" 1136768653 1300 b62933e007d01cfd073f79b963c01526 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmr9.tfm" 1136768653 1292 6b21b9c2c7bebb38aa2273f7ca0fb3af ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmsy10.tfm" 1136768653 1124 6c73e740cf17375f03eec0ee63599741 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmsy6.tfm" 1136768653 1116 933a60c408fc0a863a92debe84b2d294 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmsy9.tfm" 1136768653 1116 25a7bf822c58caf309a702ef79f4afbb ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmtt10.tfm" 1136768653 768 1321e9409b4137d6fb428ac9dc956269 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmtt8.tfm" 1136768653 768 d7b9a2629a0c353102ad947dc9221d49 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/cm/cmtt9.tfm" 1136768653 764 c98a2af25c99b73a368cf7336e255190 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/tfm/public/rsfs/rsfs10.tfm" 1229303445 688 37338d6ab346c2f1466b29e195316aa4 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/public/amsfonts/cm/cmmi10.pfb" 1248133631 36299 5f9df58c2139e7edcf37c8fca4bd384d ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/public/amsfonts/cm/cmr10.pfb" 1248133631 35752 024fb6c41858982481f6968b5fc26508 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/public/amsfonts/cm/cmsy10.pfb" 1248133631 32569 5e5ddc8df908dea60932f3c484a54c0d ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/public/amsfonts/cm/cmtt10.pfb" 1248133631 31099 c85edf1dd5b9e826d67c9c7293b6786c ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/public/amsfonts/cm/cmtt8.pfb" 1248133631 24287 6b803fa9eb1ddff9112e00519b09dd9e ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/public/amsfonts/cm/cmtt9.pfb" 1248133631 29078 718ea4567ceff944262b0f5b0800e1d9 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/urw/symbol/usyr.pfb" 1136849748 33709 b09d2e140b7e807d3a97058263ab6693 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/urw/times/utmb8a.pfb" 1136849748 44729 811d6c62865936705a31c797a1d5dada ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/urw/times/utmbi8a.pfb" 1136849748 44656 0cbca70e0534538582128f6b54593cca ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/urw/times/utmr8a.pfb" 1136849748 46026 6dab18b61c907687b520c72847215a68 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/type1/urw/times/utmri8a.pfb" 1136849748 45458 a3faba884469519614ca56ba5f6b1de1 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/ptmb7t.vf" 1136768653 1372 788387fea833ef5963f4c5bffe33eb89 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/ptmbi7t.vf" 1136768653 1384 6ac0f8b839230f5d9389287365b243c0 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/ptmr7t.vf" 1136768653 1380 0ea3a3370054be6da6acd929ec569f06 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/ptmrc7t.vf" 1136768653 1948 7330aeef3af211edff3b35fb2c12a0fd ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/ptmri7t.vf" 1136768653 1384 a9d8adaf491ce34e5fba99dc7bbe5f39 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/zptmcm7m.vf" 1136768653 1132 27520247d3fe18d4266a226b461885c2 ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/zptmcm7t.vf" 1136768653 1108 d271d6f9de4122c3f8d3b65666167fac ""
  "/usr/local/texlive/2015/texmf-dist/fonts/vf/adobe/times/zptmcm7y.vf" 1136768653 964 5673178ff30617b900214de28ab32b38 ""
  "/usr/local/texlive/2015/texmf-dist/tex/context/base/supp-pdf.mkii" 1337017135 71627 94eb9990bed73c364d7f53f960cc8c5b ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/epsf/epsf.sty" 1298075956 27123 4572e7b856182a1e6e9df1c7cc21de80 ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/etexcmds.sty" 1335995445 7612 c47308d923ec19888707b0f1792b326a ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/ifluatex.sty" 1303254447 7324 11d14f318d865f420e692d4e6c9c18c3 ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/ifpdf.sty" 1303254447 7140 ece2cc23d9f20e1f53975ac167f42d3e ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/infwarerr.sty" 1335995445 8253 3bdedc8409aa5d290a2339be6f09af03 ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/kvdefinekeys.sty" 1335995445 5152 387d9200f396b498d5fd679ae44ed898 ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty" 1335995445 14040 8de9f47fabc4ca3bd69b6d795e32751c ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/ltxcmds.sty" 1335995445 18425 775b341047ce304520cc7c11ca41392e ""
  "/usr/local/texlive/2015/texmf-dist/tex/generic/oberdiek/pdftexcmds.sty" 1335995445 19987 01cb2f3c1d21e5f05711b7fd50b17f2a ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/amsfonts/umsa.fd" 1359763108 961 6518c6525a34feb5e8250ffa91731cff ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/amsfonts/umsb.fd" 1359763108 961 d02606146ba5601b5645f987c92e6193 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/base/ifthen.sty" 1428932888 5159 bb529586d444e85eb00179221b678119 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/graphics/graphics.sty" 1428932888 14285 20d80486c4ae98139889d8789596c967 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/graphics/graphicx.sty" 1428932888 8125 557ab9f1bfa80d369fb45a914aa8a3b4 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/graphics/keyval.sty" 1428932888 2594 d18d5e19aa8239cf867fa670c556d2e9 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/graphics/trig.sty" 1428932888 3439 44bd4a1da7f27787173f2e99116b0a7f ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg" 1279039959 678 4792914a8f45be57bb98413425e4c7af ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/latexconfig/graphics.cfg" 1278958963 3563 d35e897cae3b8c6848f6677b73370b54 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty" 1303254447 12029 04d7fdf76e0464c23b5aa3a727952d7c ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/oberdiek/grfext.sty" 1335995445 7075 bd0c34fbf1ae8fd1debd2a554e41b2d5 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/oberdiek/kvoptions.sty" 1335995445 22417 c74ff4af6a1aa2b65d1924020edbbe11 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/pdftex-def/pdftex.def" 1306616590 55368 3c8a0d99822330f2dfabc0dfb09ce897 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/psnfss/omlztmcm.fd" 1137110629 411 12564a37a279e4e0b533cdf5e03eeb7c ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/psnfss/omsztmcm.fd" 1137110629 348 f4ce75d394e7d9ac12ca7aac4045ed77 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/psnfss/omxztmcm.fd" 1137110629 329 c8cddcc90b6f567b28408eb374773c9c ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/psnfss/ot1ptm.fd" 1137110629 961 15056f4a61917ceed3a44e4ac11fcc52 ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/psnfss/ot1ztmcm.fd" 1137110629 329 aee7226812ba4138ac67a018466b488d ""
  "/usr/local/texlive/2015/texmf-dist/tex/latex/rotating/rotating.sty" 1238284825 7131 a49b179ae7108fe42935176733099fc0 ""
  "/usr/local/texlive/2015/texmf-dist/web2c/texmf.cnf" 1428852030 31983 18aeec08f48baa3cf3508d538dd5f459 ""
  "/usr/local/texlive/2015/texmf-var/fonts/map/pdftex/updmap/pdftex.map" 1432479641 1764542 91654ba72bdb667e84ea24f076ae16d8 ""
  "/usr/local/texlive/2015/texmf-var/web2c/pdftex/pdflatex.fmt" 1465224599 3849858 69cc909b429df4297174ef15d9d6f231 ""
  "/usr/local/texlive/2015/texmf.cnf" 1432479636 577 0d611272082f3cdb8e80e09a3c69cf07 ""
  "amsfonts.sty" 1475862540 7992 7a12630bc841f1c895895a7b9763b259 ""
  "bm.sty" 1475862540 11793 e48c43ab20c471f6925111e6c631cef1 ""
  "jfp1.cls" 1475862541 54212 7b8c4c69b21f0acae22aac133e848a23 ""
  "jfp2egui.aux" 1476427432 4374 b492f61bb701cc6c240141b1cbae15cd ""
  "jfp2egui.tex" 1475862542 50885 960cf7d22664b0bae90fabd4788dfb5c ""
  "jfp2egui.toc" 1476427432 2068 35240358a0b649627a32da5ddb83336e ""
  "mathptmx.sty" 1475862542 1506 f999f79c035add97f92146f6c9a4ba79 ""
  (generated)
  "jfp2egui.pdf"
  "jfp2egui.log"
  "jfp2egui.aux"
  "jfp2egui.toc"
```

**Comment by yitzchak** [comment]
@alexander-myltsev Please delete the DVI file, do a clean and then a rebuild.

**Comment by alexander-myltsev** [comment]
Worked out, cool.

**Comment by baubaid** [comment]
I seem to be having the same (or a similar) issue. Reverse searching from the PDF (I'm using Skim) works fine and takes me to the right location in Atom.

However, when I use the ctrl-alt-S keybinding to do forward search, the PDF in Skim seems to refresh but I'm not taken to the correct location in the source code, and the PDF stays at the same location it was before.

This issue also persists whether Skim already open with the file or not.

**Comment by yitzchak** [comment]
@baubaid This is a know issue with Skim and should be fixed in their next version.

**Comment by yitzchak** [comment]
@karyon Are you still seeing this issue? There is a new version of pdf-view out (v0.51.0), btw.

**Comment by karyon** [comment]
@yitzchak right the only thing i get is `No opener found that can open C:/path/to/thesis.dvi`. This file does not even exist, output is set to pdf. using atom-latex 0.39.1.

**Comment by yitzchak** [comment]
@karyon Which latex engine are you using?

**Comment by karyon** [comment]
xelatex.

**Comment by karyon** [comment]
...and clean seems to do nothing. i re-checked, yes i have 0.39.1 installed. pretty sure that works on my other machine.

**Comment by yitzchak** [comment]
MiKTeX or TeX Live?

**Comment by karyon** [comment]
miktex. unfortunately, after some more playing around, both building and clean work again. i cannot say what exactly i changed to make it work.

yeah so syncing-after-building with pdf-view is still kinda buggy. what i see:

i have two panes open, one my tex-file, one pdf-view with the output. when i edit something and build, it compiles, and when the compiling has finished ("latex build" gets replaced by "latex log") syncs to the cursors position but seem to have the old output still loaded. then a second later it jumps to some seemingly random position in the document, then 5 seconds later (atom fully utilizes a core during that, i think this is pdf-view loading the pdf) it syncs to the correct location.

also, when the pdf-view pane is not open and i do the sync command, it opens pdf-view but stays at the first page.

also, when closing the pdf-view pane, choosing sumatra and compiling, atom still uses the CPU to 100% for a few seconds after compiling, just as if pdf-view would still be running. haven't investigated further.

also, the editor crashed sometimes (but not consistently, maybe every fifth time) after compiling. sadly atom doesn't give any more information, so i can't say what exactly crashed.

maybe the long loading times of pdf-view screw things up. fwiw here's the document: https://github.com/karyon/masterthesis should compile out of the box with xelatex.

**Comment by yitzchak** [comment]
@karyon I just reproduced on Arch/TeX Live.

**Comment by karyon** [comment]
glad to hear that!

**Comment by yitzchak** [comment]
Including the crash :disappointed:

**Comment by yitzchak** [comment]
@karyon Hopefully I just found it. If you want you can try it out by changing

```javascript
    atom.workspace.open(filePath, {'split': 'right'}).then(forwardSync)
```

in `lib/opener/pdf-open-opener.js` to

```javascript
    atom.workspace.open(filePath, {'split': 'right'}).then(() => forwardSync(filePath, texPath, lineNumber))
```

**Comment by karyon** [comment]
(i suppose you meant `pdf-view-opener.js`)

i edited that file, restarted atom (i assumed that the code is not cached anywhere) and got the exact same behaviour. syncing when no pdf pane was open still opened it but stayed at the first page, and the first build i did directly crashed, the second build first jumped to a weird position before jumping to the correct one.

**Comment by yitzchak** [comment]
Looks like there is multiple problems going on. I think pdf-view doesn't like to be told to sync when it is still loading the document.

**Comment by yitzchak** [comment]
@karyon Quick update: this looks to be a loading issue in pdf-view, the SyncTeX parsing is working, but calling forward sync right after loading will not work. Delaying for a few seconds appears to work, probably due to your document size.

**Comment by yitzchak** [comment]
I have reproduced the error with a small document. It is a bug in `pdf-view`. They are loading the document with promises/callbacks but resetting their own `this.updating` flag right away, plus not checking the flag before a scroll. I'll submit a PR to them soon.

**Comment by yitzchak** [comment]
PR izuzak/atom-pdf-view#157

**Comment by yitzchak** [comment]
@karyon The PR just got merged at pdf-view. If the document is very large there is still a delay until sync happens when the document is first loaded. For your thesis it took about 12 seconds on my system and it is no slouch.

**Comment by karyon** [comment]
now syncing does not work at all with atom-latex 0.40 and pdf-view 0.52 :( not right after building and not even with ctrl-alt-s, and backwards sync from pdf-view does not work either. with sumatra, everything works as expected.

also just a heads up, with atom 0.13.0 i get some deprecation warnings, one of them comes from atom-latex.

**Comment by thomasjo** [comment]
> also just a heads up, with atom 0.13.0 i get some deprecation warnings, one of them comes from atom-latex.

We already have a [PR](https://github.com/thomasjo/atom-latex/pull/267) for that — thanks for the heads up regardless.

**Comment by yitzchak** [comment]
@karyon If you are still using TeX Live's `synctex` binary with MiKTeX you may need to tell pdf-view about that. There is a setting `pdf-view.syncTeXPath` in the pdf-view settings page now. I had to set the *full* path to `synctex` to get it to work.

**Comment by karyon** [comment]
yeah i tried on a machine where i actually didn't have a synctex binary.

on a machine that has one, ctrl-alt-s still does nothing, ctrl-alt-b does compile the document but then says `Cannot read property 'find' of undefined` and does not open the pdf. backward sync from pdf-view to the code worked as long as i had the pdf pane open, now i closed it and both build and sync don't bring it back. choosing sumatra as viewer doesn't work either, sumatra is never opened. i tried a clean build, changed nothing.

i hope this is not one more problem caused by my confusion...

btw i noticed that when switching the viewer, a full rebuild seems to be performed.

**Comment by yitzchak** [comment]
@karyon Have you done a clean and does `foo-thesis.fdb_latexmk` exist?

**Comment by karyon** [comment]
yes, i have done a clean and that file exists. on another machine with the same setup i don't have this problem, but i can't say what the difference between the machines might be. any tips where to look would be appreciated.

**Comment by yitzchak** [comment]
@karyon Let's do a version check on the machine that is not working.

What is the version of Atom and atom-latex? Then set your logging level to `info` in the `latex` settings page and run `latex:check-runtime`. What does it report in the log?

**Comment by karyon** [comment]
atom: 1.13.0-beta2
atom-latex: 0.40.0

i seem to be unable to copy anything from the log pane (should i open an issue for that?), so here's a screenshot:
![2016-11-17 14_03_49](https://cloud.githubusercontent.com/assets/1891915/20390336/b43851ae-acce-11e6-917d-9060fe9eb0e7.png)

again to be clear, building succeeds and the resulting pdf is correct, but after building atom-latex shows `Cannot read property 'find' of undefined` instead of opening a viewer, and syncing does not work in any direction.

**Comment by yitzchak** [comment]
We haven't added the ability to copy the log to the clipboard yet. There is a PR pending #289 for it though.

Are you testing a beta of `latexmk`?

**Comment by karyon** [comment]
that's exactly what's happening, i still had a beta latexmk.pl in place that the author sent me when i suggested #255 to him, and i had forgotten about that. i reverted it to the old version and now that error is gone, the pdf-viewer is opening and syncing works.

sorry for taking so much of your time.

regarding the original issue, pdf-view is still behaving weirdly as it still jumps around when the pdf is recompiled, but i've seen this during startup of atom as well, so i guess this is entirely pdf-view's fault.

the editor still occasionally crashes when opening a pdf (in both pdf-view and sumatra), and sometimes the editor goes unresponsive with a perl process using 100% of the cpu, but both are not reliably reproducable and i don't know what triggers them.

one thing that i mentioned above: changing the pdf viewer seems to cause the next build to be a full rebuild. don't know if that is intended.

as far as i'm concerned this issue here can be closed.

**Comment by yitzchak** [comment]
@karyon No worries about time invested!

Rebuild is triggered up any change to `latex` settings. I'll look and see if we can make it more specific.

Has John Collins given you an estimated release time for the next version? I am concerned that we have a breaking issue approaching.

**Comment by karyon** [comment]
no, he has not. i can give you the version he sent me if you want, but maybe it's better if you contact him directly. i haven't found the place where the code is hosted or an issuetracker or anything like that.

**Comment by yitzchak** [comment]
@karyon Sounds good. Thanks for the info.
